src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs

```diff
@@ -23,6 +20,11 @@ public abstract class SyntaxGenerator : ILanguageService
     {
         public static SyntaxRemoveOptions DefaultRemoveOptions = SyntaxRemoveOptions.KeepUnbalancedDirectives | SyntaxRemoveOptions.AddElasticMarker;
 
+        //// Use an internal constructor so this is only derivable by our own types.
+        internal SyntaxGenerator()
```

Comment:
I think the compat risk is low here, and getting this in earlier is better rather than later.